community/kernel/src/main/java/org/neo4j/kernel/api/txtracking/TransactionIdTracker.java

```diff
@@ -109,6 +110,6 @@ private boolean isReady( long oldestAcceptableTxId ) throws TransactionFailureEx
      */
     public long newestEncounteredTxId()
     {
-        return transactionIdStore.getLastClosedTransactionId();
+        return transactionIdStoreSupplier.get().getLastClosedTransactionId();
```

Comment:
Maybe extract `transactionIdStoreSupplier.get()` in a method with a comment saying why is it required.